src/Microsoft.IdentityModel.Clients.ActiveDirectory/Features/CertificateAssertion/ClientAssertionCertificate.cs

```diff
@@ -50,12 +51,12 @@ public ClientAssertionCertificate(string clientId, X509Certificate2 certificate)
         {
             if (string.IsNullOrWhiteSpace(clientId))
             {
-                throw new ArgumentNullException("clientId");
+                throw new ArgumentNullException(nameof(clientId));
             }
 
             if (certificate == null)
             {
-                throw new ArgumentNullException("certificate");
+                throw new ArgumentNullException(nameof(certificate));
```

Comment:
You could write:
this.Certificate = certificate ?? throw new ArgumentNullException(nameof(certificate))